OrbitCore/LinuxPerfUtils.h

```diff
@@ -9,51 +9,118 @@
 
 #include <asm/unistd.h>
 #include <asm/perf_regs.h>
+#include <cerrno>
+#include <cstdint>
 #include <linux/perf_event.h>
 #include <linux/version.h>
-#include <sys/errno.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
+#include <unistd.h>
 
 //-----------------------------------------------------------------------------
 namespace LinuxPerfUtils
 {
-    // TODO: these constants should be made available in some settings
-    static constexpr uint32_t STACK_SIZE = (1u << 13u) - 8;
-    // Sample stack- and instruction pointer (used later for stack unwinding)
-    static constexpr uint64_t SAMPLE_REGS_USER = (0x1 << PERF_REG_X86_SP) | (0x1 << PERF_REG_X86_IP);
-
-    // This must be in sync with the perf_event_record and the perf_sample_id stucts.
-    static constexpr uint64_t SAMPLE_TYPE_FLAGS = 
-        PERF_SAMPLE_TID | 
-        PERF_SAMPLE_TIME | 
-        PERF_SAMPLE_CPU | 
-        0u;
+    // This must be in sync with the perf_event_record and the perf_sample_id structs.
+    static constexpr uint64_t SAMPLE_TYPE_FLAGS =
+        PERF_SAMPLE_TID |
+        PERF_SAMPLE_TIME |
+        PERF_SAMPLE_CPU;
+
+    // TODO: These constants should be made available in some settings.
+    // Max to pass to perf_event_open without getting an error is (1u << 16u) - 8,
+    // because the kernel stores this in a short and because of alignment reasons.
+    // But the size the kernel actually returns is smaller, because the maximum
+    // size of the entire record the kernel is willing to return is (1u << 16u) - 8.
+    // If we want the size we pass to concide with the size we get, we need to
+    // pass a lower value. For the current layout of perf_record_with_stack,
+    // the maximum size is 65312, but let's leave some extra room.
```

Comment:
So, this can change depending on kernel changes? Is there a programmatic way to get the correct maximum size?

Reply:
I don't know a programmatic way and also I doubt this could change, or at least that it could shrink.

Reply:
The reason for this is that the header for perf_record has _u16 size field, I do not think this is something that can change without introducing new syscall.